roles_revamp.md

```diff
@@ -0,0 +1,37 @@
+# Proposal: Revamp Roles
+
+*Author*: Brian Ccoai <@bcoca>
+
+*Date*: 04/01/2016
+
+- Status: New
+- Proposal type: core design
+- Targeted Release: 2.2
+- PR for Comments: 
+- Estimated time to implement: 4 weeks
+
+
+## Motivation
+Roles are a good way to share and reuse resources, but they are a bit limited in application.
+This proposal aims at expanding the usefulness and reusability of roles.
+
+### Problems
+Roles lack flexibility in execution and inclusion.
+
+## Solution proposal
+- Add exec_main flag to roles (default True to keep current behaviour), when false a role will not execute tasks/main.yml in `roles:` section of a play.
```

Comment:
it seems like exec_main: is usage-time whereas deciding to put in a main.yml or not is definition-time. ie:  If we want the user of a role to be able to choose whether to execute tasks/main.yml whether or not the author of the role put it in, then we'd need to have something here.